test/e2e/empty_storageclass_test.go

```diff
@@ -0,0 +1,56 @@
+package e2e
+
+import (
+	_ "embed"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	//go:embed testdata/empty_storageclass/manifest.yaml
+	emptyStorageClassManifest string
+)
+
+const nsEmptyStorageClassTest = "empty-storageclass-test"
+
+func testEmptyStorageClass() {
+	var cc CleanupContext
+	BeforeEach(func() {
+		createNamespace(nsEmptyStorageClassTest)
+		cc = commonBeforeEach()
+	})
+	AfterEach(func() {
+		_, err := kubectl("delete", "namespaces", nsEmptyStorageClassTest)
+		Expect(err).ShouldNot(HaveOccurred())
+		_, err = kubectl("delete", "pv", "test-empty-storageclass")
+		Expect(err).ShouldNot(HaveOccurred())
+		commonAfterEach(cc)
+	})
+
+	It("should create a Pod with empty StorageClass PVC", func() {
+		By("applying a Pod with empty StorageClass PV/PVC")
+		yaml := []byte(fmt.Sprintf(emptyStorageClassManifest, nsEmptyStorageClassTest, nsEmptyStorageClassTest))
+		_, err := kubectlWithInput(yaml, "apply", "-f", "-")
+		Expect(err).ShouldNot(HaveOccurred())
+
+		By("checking the Pod is running")
+		Eventually(func() error {
+			var pod corev1.Pod
+			err := getObjects(&pod, "pod", "-n", nsEmptyStorageClassTest, "test-empty-storageclass")
+			if err != nil {
+				return err
+			}
+
+			for _, cond := range pod.Status.Conditions {
+				if cond.Type == corev1.PodReady && cond.Status == corev1.ConditionTrue {
+					return nil
+				}
+			}
+
+			return nil
```

Comment:
Okay, here we should return an error, otherwise it effectively only checks that we can get the Pod definition and L47-51 has no meaning :(